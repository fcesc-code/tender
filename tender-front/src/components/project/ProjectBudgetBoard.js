import React from 'react';
import { Link } from 'react-router-dom';


function ProjectBudgetBoard(props) {
  const project = props.project;
  return (
    <div className="project__budgetBoard">
      <p>Some data of the board, create table</p>
    </div>
  );
}

export default ProjectBudgetBoard;









