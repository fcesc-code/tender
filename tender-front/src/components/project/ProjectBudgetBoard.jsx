import React from 'react';
import { connect } from 'react-redux';
import './projectBudgetBoard.sass';
import Spinner from '../common/Spinner';

function ProjectBudgetBoard( { budget } ) {
  console.log(' the budget inside the compo is ', budget);
  for (let p in budget){ console.log('prop ', p)};

  return (
    ( Object.keys(budget).length === 0 ) ? (<Spinner/>) : (
      <div className="project__budgetBoard">
        <table className='board__table'>
          <thead>
            <tr>
              {budget.data.headers.map((header, index)=>{
                return(<th key={`H0.${index}`}>{header}</th>)
              })}
            </tr>
          </thead>
          <tbody>
            {budget.data.entries.map((entry, rowIndex)=>{
              return(<tr key={rowIndex}>{entry.map((field, index)=>{
              return(<td key={`R${rowIndex}.${index}`}>{field}</td>)
              })}</tr>)
            })}
          </tbody>
        </table>
      </div>
    )
  );
}

function mapStateToProps( state, { view } ){
  const selectedBudget = (Object.keys(state.project.data).length === 0) ? '' : ( view === '' ) ? state.project.data[0] : state.project.data.filter(item=>item._id === view)[0];
  console.log('showing insides of mapStateToProps with selectedBudget', view, selectedBudget);
  return {
    budget: selectedBudget
  }
}

export default connect(mapStateToProps)(ProjectBudgetBoard);









