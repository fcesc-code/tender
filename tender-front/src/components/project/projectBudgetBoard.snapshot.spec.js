import React from "react";
import renderer from "react-test-renderer";
import ProjectBudgetBoard from "./ProjectBudgetBoard.jsx";

const tree = renderer.create(<ProjectBudgetBoard />).toJSON();

describe("ProjectBudgetBoard component snapshot", () => {
  test("should match", () => {
    expect(tree).toMatchSnapshot();
  });
});
