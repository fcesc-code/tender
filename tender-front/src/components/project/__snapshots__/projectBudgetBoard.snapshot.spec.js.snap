// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectBudgetBoard component snapshot should match 1`] = `
<div
  className="project__budgetBoard"
>
  <table
    className="board__table"
  >
    <thead>
      <tr>
        <th>
          ref
        </th>
        <th>
          description
        </th>
        <th>
          unit
        </th>
        <th>
          quantity
        </th>
        <th>
          unit_price
        </th>
        <th>
          amount
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          d934laa
        </td>
        <td>
          Dummy position
        </td>
        <td>
          m
        </td>
        <td>
          198.12
        </td>
        <td>
          2030
        </td>
        <td>
          402184
        </td>
      </tr>
      <tr>
        <td>
          d93y900
        </td>
        <td>
          Dummy position
        </td>
        <td>
          m2
        </td>
        <td>
          10.5
        </td>
        <td>
          12410
        </td>
        <td>
          130305
        </td>
      </tr>
      <tr>
        <td>
          ds64ls7
        </td>
        <td>
          Dummy position
        </td>
        <td>
          m
        </td>
        <td>
          340
        </td>
        <td>
          536
        </td>
        <td>
          185456
        </td>
      </tr>
      <tr>
        <td>
          5934ls3
        </td>
        <td>
          Dummy position
        </td>
        <td>
          m
        </td>
        <td>
          28
        </td>
        <td>
          1374
        </td>
        <td>
          38742
        </td>
      </tr>
      <tr>
        <td>
          r935ls3
        </td>
        <td>
          Dummy position
        </td>
        <td>
          m2
        </td>
        <td>
          4.3
        </td>
        <td>
          1000
        </td>
        <td>
          4300
        </td>
      </tr>
      <tr>
        <td>
          dav57s3
        </td>
        <td>
          Dummy position
        </td>
        <td>
          m
        </td>
        <td>
          1
        </td>
        <td>
          208500
        </td>
        <td>
          208500
        </td>
      </tr>
      <tr>
        <td>
          2934ls6
        </td>
        <td>
          Dummy position
        </td>
        <td>
          m
        </td>
        <td>
          125
        </td>
        <td>
          260
        </td>
        <td>
          32500
        </td>
      </tr>
      <tr>
        <td>
          d2835s3
        </td>
        <td>
          Dummy position
        </td>
        <td>
          m
        </td>
        <td>
          56
        </td>
        <td>
          956
        </td>
        <td>
          53536
        </td>
      </tr>
      <tr>
        <td>
          h934fas
        </td>
        <td>
          Dummy position
        </td>
        <td>
          m
        </td>
        <td>
          11
        </td>
        <td>
          125
        </td>
        <td>
          1375
        </td>
      </tr>
      <tr>
        <td>
          gfrt6s3
        </td>
        <td>
          Dummy position
        </td>
        <td>
          m
        </td>
        <td>
          12.5
        </td>
        <td>
          5634
        </td>
        <td>
          70425
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
